tempodb/blocklist/poller.go

```diff
@@ -360,3 +468,19 @@ func sumTotalBackendMetaMetrics(blockMeta []*backend.BlockMeta, compactedBlockMe
 		compactedBlockMetaTotalBytes:   sumTotalBytesCBM,
 	}
 }
+
+func uuidInBlocklist(id uuid.UUID, metas []*backend.BlockMeta, compactedMetas []*backend.CompactedBlockMeta) bool {
+	for _, i := range metas {
```

Comment:
the thing that's being passed in to the `Do()` function is a `*List` right? the list object has maps to meta and compactedMetas which would be much faster lookup than this.

why don't we just take a *List instead of an interface?